Shared/Samples/Monitor changes to layer view state/MonitorChangesToLayerViewStateView.swift

```diff
@@ -0,0 +1,122 @@
+            center: Point(x: -13e6, y: 51e5, spatialReference: .webMercator),
+            scale: 2e7
+        )
+        return map
+    }()
+    
+    /// The Satellite (MODIS) Thermal Hotspots and Fire Activity feature layer.
+    @State private var featureLayer: FeatureLayer = {
+        let portalItem = PortalItem(url: .thermalHotspotsAndFireActivity)!
+        let featureLayer = FeatureLayer(item: portalItem)
+        
+        featureLayer.minScale = 1e8
+        featureLayer.maxScale = 6e6
+        
+        return featureLayer
+    }()
+    
+    /// The text representing the view status of the feature layer.
+    @State private var statusText = ""
+    
+    /// A Boolean value indicating whether the feature layer is visible.
+    @State private var layerIsVisible = true
+    
+    var body: some View {
+        MapView(map: map)
+            .onLayerViewStateChanged { layer, layerViewState in
+                // Only checks the view state of the feature layer.
+                guard layer.id == featureLayer.id else { return }
+                
+                // Gets the status from the view state for the layer.
+                statusText = layerViewState.status.label
+            }
+            .overlay(alignment: .top) {
+                Text("Layer view status:\n\(statusText)")
+                    .multilineTextAlignment(.center)
+                    .frame(maxWidth: .infinity, alignment: .center)
+                    .padding(8)
+                    .background(.regularMaterial, ignoresSafeAreaEdges: .horizontal)
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Toggle(
+                        layerIsVisible ? "Layer Enabled" : "Layer Disabled",
+                        isOn: $layerIsVisible
+                    )
+                    .onChange(of: layerIsVisible) { newValue in
+                        featureLayer.isVisible = newValue
+                    }
+                }
+            }
+            .onAppear {
+                // Adds the feature layer to the map.
+                map.addOperationalLayer(featureLayer)
+            }
+    }
+}
+
+private extension LayerViewState.Status {
+    /// A human-readable label for the status.
+    var label: String {
+        var statuses: [String] = []
+        if self.contains(.active) {
+            statuses.append("Active")
+        }
+        if self.contains(.notVisible) {
+            statuses.append("Not Visible")
+        }
+        if self.contains(.outOfScale) {
+            statuses.append("Out of Scale")
+        }
+        if self.contains(.loading) {
+            statuses.append("Loading")
+        }
+        if self.contains(.error) {
+            statuses.append("Error")
+        }
+        if self.contains(.warning) {
+            statuses.append("Warning")
+        }
+        
+        if !statuses.isEmpty {
+            return statuses.joined(separator: ", ")
```

Comment:
In the spirit of modeling best practices for our customers, instead of joining the strings, you could return an array of labels and then use a list format style:

```swift
Text(
    """
    Layer view status:
    \(layerStatus.labels, format: .list(type: .and, width: .narrow))
    """
)
```

As demonstrated in that snippet, you could store the status itself rather than the formatted string. By doing so, you could check that the new status is different than the old status before updating it, making view updates more efficient.

Reply:
Issue for this here: #400